source/simulator/helper_functions.cc

```diff
@@ -545,7 +545,7 @@ namespace aspect
 
     // Do a checkpoint if this is the end of simulation,
     // and the termination criteria say to checkpoint at the end.
-    if (termination_output.first && termination_output.second)
+    if (force_writing_checkpoint)
       write_checkpoint = true;
```

Comment:
I think this would be easier to read if you did
```
  bool write_checkpoint = force_writing_checkpoint;
  if (write_checkpoint == false && condition 1)
    ...
  if (write_checkpoint == false && condition 2)
    ...
```
This way you save a bit of time by not executing some checks, and conceptually you start with the "force" operation.